packages/commons-server/src/libs/server/crud.ts

```diff
@@ -192,10 +205,12 @@ export const databucketActions = (
     case 'create': {
       if (Array.isArray(databucket.value)) {
         if (typeof requestBody === 'object' && requestBody != null) {
-          requestBody = {
-            [crudRouteParamName]: generateUUID(),
-            ...requestBody
-          };
+          if (routeCrudKey === 'id') {
+            requestBody = {
+              [routeCrudKey]: generateUUID(),
+              ...requestBody
+            };
+          }
```

Comment:
Why limit the default uuid generation to `id`? This behavior is documented and should stay the same I think no matter the chosen key.